lib/TippyTopProvider.js

```diff
@@ -0,0 +1,67 @@
+/* globals require, JSON */
+"use strict";
+
+const data = require("sdk/self").data;
+const URL = require("sdk/url").URL;
+
+const DEFAULT_OPTIONS = {
+  sites: null
+};
+
+function TippyTopProvider(options = {}) {
+  this.options = Object.assign({}, DEFAULT_OPTIONS, options);
+  this.init();
+}
+
+TippyTopProvider.prototype = {
+  /**
+   * Initialize the sites object
+   */
+  init() {
+    // If the sites were passed into the constructor (for testability), use those.
+    // Else, read them from top_sites.json.
+    let sites;
+    if (this.options.sites) {
+      sites = this.options.sites;
+    } else {
+      sites = JSON.parse(data.load("content/favicons/top_sites.json"));
```

Comment:
Ah yes. This is a bit more verbose, but we can wire it up during the `npm run bundle` step:

```js
const writeFile = require("fs").writeFileSync;
const join = require("path").join;
const tippy = require("tippy-top-sites");

const OUT_FILE = join("data", "content", "favicons", "top_sites.json");

writeFile(OUT_FILE, JSON.stringify(tippy));
```